nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+					Description: "Uplink name of VMware vSphere Distributed Switch (VDS)",
+				},
+			},
+		},
+	}
+}
+
+func getHostSwitchProfileIDsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Identifiers of host switch profiles to be associated with this host switch",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"key": {
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(hostSwitchProfileIDKeyValues, false),
+					Required:     true,
+				},
+				"value": {
+					Type:        schema.TypeString,
+					Description: "key value",
+					Required:    true,
+				},
+			},
+		},
+	}
+}
+
+func getIPAssignmentSchema(varPath string) *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Specification for IPs to be used with host switch virtual tunnel endpoints",
+		MaxItems:    1,
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"assigned_by_dhcp": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					//TODO: as ExactlyOneOf cannot be used, use some alternate validation
+					//ExactlyOneOf: []string{
+					//	varPath + ".0.static_ip_list_spec",
+					//	varPath + ".0.static_ip_mac_list_spec",
+					//	varPath + ".0.static_ip_pool_spec",
+					//},
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:        schema.TypeBool,
+								Required:    true,
+								Description: "Enables DHCP assignment. Should be set to true",
+							},
+						},
+					},
+				},
+				"static_ip_list_spec": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_list": {
+								Type:        schema.TypeList,
+								Description: "List of IPs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Schema{
+									Type:         schema.TypeString,
+									ValidateFunc: validateSingleIP(),
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_mac_list_spec": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_mac_list": {
```

Comment:
this would be sub-clause, not a list

Reply:
Why? Per the [spec,](https://vdc-download.vmware.com/vmwb-repository/dcr-public/201a09cb-5610-49e6-9d31-7db0920e3a65/65f199d6-8516-4092-9b19-906559f972b5/api_includes/types_StaticIpMacListSpec.html) it could contain 32 items

Reply:
Sorry, I didn't phrase this well. For hcl perspective, this would be a subclause rather than a list: 
```ip_mac {
    ...
   }```
so we shouldn't call it `list`. Maybe `ip_mac` or `ip_mac_pair`